File: Classes/Script/python.site.prv/nnt.modules.m

# import "Core.h"
# import "pynnt.h"
# import "nnt.modules.h"
# import "nnt.core.h"

NNT_BEGIN_OBJC

static PyObject* _info(PyObject *hdl, PyObject *args) {
    return Py_BuildValue("s", "Python Bridge of NNT. 2011.");
}

static PyMethodDef _methods[] = {
    {"info",  _info, METH_VARARGS, "Get info for PyNNT."},
    {NULL, NULL, 0, NULL}
};

static PyObject *_exception = nil;

void PyNNT_LoadModules() {
    PyObject* m = Py_InitModule("nnt", _methods);
    
    if (m == NULL)
        return;
    
    _exception = PyErr_NewException("nnt.error", NULL, NULL);
    Py_INCREF(_exception);
    PyModule_AddObject(m, "error", _exception);
    
    // load modules.
    // load core module.
    PyNNT_LoadCoreModule(m);
}

NNT_END_OBJC

File: Classes/Script/python.site.prv/nnt.modules.cpp

# include "Core.h"
# include "pynnt.h"
# include "nnt.modules.h"
# include "nnt.core.h"

NNT_BEGIN_C

static PyObject* pynnt_test(PyObject*, PyObject*)
{
    trace_msg("call pynnt test function");
    return Py_BuildValue("s", "Nnt + Python is ok");
}

static PyMethodDef gs_methods[] = {
    {"test", pynnt_test, METH_NOARGS, "test"},
    {0, 0, 0, 0}
};

static PyObject* pynnt_error = NULL;

void PyNnt_LoadModules_c()
{
    PyObject* m = Py_InitModule("nnt", gs_methods);
    if (m == NULL)
        return;

    pynnt_error = PyErr_NewException(pystr("nnt.error"), NULL, NULL);
    Py_INCREF(pynnt_error);
    PyModule_AddObject(m, "error", pynnt_error);

    PyNnt_LoadCoreModule(m);
}

NNT_END_C
